### PR TITLE
Parser integration, handle nested()

### PR DESCRIPTION
**Pull request by jinglongwu**

### CODE REVIEW

File: parser.py [hunk 1]
```diff
@@ -52,7 +74,7 @@ def parse(self, content, root=None):
         If the root is not given, new root will created and returned.
         """
         if root is None:
-            root = Element(None)
+            root = Element(content)
```
Comment (andysim3d) [inline parser.py:77]: IIUC, the root is used as a placeholder, why we put content in it?

File: parser.py [hunk 2]
```diff
@@ -97,3 +119,45 @@ def parse(self, content, root=None):
                 self.parse(child.content(), child)
 
         return root
+
+def create_paragrah_and_block_parsers():
+    paragraph_elements = [
+        Header,
+        HorizontalRule,
+        ListParagraph,
+        QuoteParagraph,
```
Comment (andysim3d) [inline parser.py:134]: also fenced code.

File: parser.py [hunk 3]
```diff
@@ -97,3 +119,45 @@ def parse(self, content, root=None):
...
+        CodeBlock,
+        FenchedCodeBlock,
+        StrikethroughBlock,
+    ]
+    block_functors = []
+    for element in block_elements:
+        parser = element.parse
+        block_functors.append(parser)
+
+    p_parser = ParagraphParser()
```
Comment (andysim3d) [inline parser.py]: paragraph_functors and block_functors seems useless here.

File: parser.py [hunk 4]
```diff
@@ -97,3 +119,45 @@ def parse(self, content, root=None):
                 self.parse(child.content(), child)
```
Comment (andysim3d) [inline parser.py]: also, if a child is not "nested", we should skip that child's recursive parse.

**Review by andysim3d — COMMENTED** [review]
basic LGTM.

File: parser.py (hunk 1 above)
Comment (jinglongwu) [reply parser.py:77]: let's leave it atm for the sake of debugging purpose, once our code tend to be stable, we will remove it for saving memory

**Review by andysim3d — APPROVED** [review]
LGTM

**Pull request merged**